Core/tests/Unit/RequestWrapperTest.php

```diff
@@ -488,6 +488,53 @@ public function testDefaultToAnonymousCredentialsWhenNoOthersExist()
         $this->assertInstanceOf(FetchAuthTokenInterface::class, $fetcher);
         $this->assertNull($fetcher->fetchAuthToken()['access_token']);
     }
+
+    public function testSetsQuotaProjectOnCredentialsWithKeyFile()
+    {
+        $quotaProject = 'test-quota-project';
+        $requestWrapper = new RequestWrapper([
+            'quotaProject' => $quotaProject,
+            'keyFile' => json_decode(file_get_contents(Fixtures::JSON_KEY_FIXTURE()), true)
+        ]);
+
+        $this->assertEquals(
+            $quotaProject,
+            $requestWrapper->getCredentialsFetcher()->getQuotaProject()
+        );
+    }
+
+    public function testSetsQuotaProjectOnCredentialsWithADC()
+    {
+        $quotaProject = 'test-quota-project';
+        $keyFilePath = Fixtures::JSON_KEY_FIXTURE();
+        putenv("GOOGLE_APPLICATION_CREDENTIALS=$keyFilePath");
+        $requestWrapper = new RequestWrapper([
+            'quotaProject' => $quotaProject
+        ]);
+
+        $this->assertEquals(
+            $quotaProject,
+            $requestWrapper->getCredentialsFetcher()->getQuotaProject()
+        );
+    }
+
+    public function testAddsQuotaProjectHeaderToRequest()
+    {
+        $quotaProject = 'test-quota-project';
+        $requestWrapper = new RequestWrapper([
+            'quotaProject' => $quotaProject,
+            'httpHandler' => function ($request, $options = []) use ($quotaProject) {
+                $userProject = $request->getHeaderLine('X-Goog-User-Project');
+                $this->assertEquals($quotaProject, $userProject);
+                return new Response(200);
+            },
+            'accessToken' => 'abc'
+        ]);
+
+        $requestWrapper->send(
+            new Request('GET', 'http://www.example.com')
+        );
+    }
```

Comment:
nit: Add a test for the use-case I describe above: When `quotaProject` is set on both ADC and a client option

Reply:
Test added 👍